src/c++/perf_analyzer/infer_context.cc

```diff
@@ -66,16 +66,17 @@ InferContext::SendSequenceInferRequest(uint32_t seq_stat_index, bool delayed)
 {
   // Need lock to protect the order of dispatch across worker threads.
   // This also helps in reporting the realistic latencies.
-  std::lock_guard<std::mutex> guard(sequence_stat_[seq_stat_index]->mtx_);
-  if (!early_exit && !sequence_stat_[seq_stat_index]->paused_) {
-    SetInferSequenceOptions(seq_stat_index, options_);
+  std::lock_guard<std::mutex> guard(
+      sequence_manager_->GetMutex(seq_stat_index));
+  if (!early_exit && execute_) {
+    sequence_manager_->SetInferSequenceOptions(seq_stat_index, options_);
 
     // Update the inputs if required
     if (using_json_data_) {
-      UpdateSeqJsonData(sequence_stat_[seq_stat_index]);
+      UpdateSeqJsonData(seq_stat_index);
     }
 
-    sequence_stat_[seq_stat_index]->remaining_queries_--;
+    sequence_manager_->GetRemainingQueries(seq_stat_index)--;
```

Comment:
I don't like returning a reference to a member variable here.  Create a function to decrement count.

Reply:
Added `DecrementRemainingQueries()` and `SetRemainingQueries()` for when remaining queries gets reset to 1 [here]() 👍